test/audits/uncontrolled-tabpanel-test.js

```diff
@@ -0,0 +1,132 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+module('UncontrolledTabpanel');
+
+test('No elements === no problems.', function() {
+  // Setup fixture
+  var fixture = document.getElementById('qunit-fixture');
+  deepEqual(
+    axs.AuditRules.getRule('uncontrolledTabpanel').run({ scope: fixture }),
+    { result: axs.constants.AuditResult.NA }
+  );
+});
+
+test('No roles === NA.', function() {
+  // Setup fixture
+  var fixture = document.getElementById('qunit-fixture');
+  for (var i = 0; i < 10; i++)
+    fixture.appendChild(document.createElement('div'));
+
+  deepEqual(
+    axs.AuditRules.getRule('uncontrolledTabpanel').run({ scope: fixture }),
+    { result: axs.constants.AuditResult.NA }
+  );
+});
+
+test('No elements with role tabpanel === NA.', function() {
+  // Setup fixture
+  var fixture = document.getElementById('qunit-fixture');
+  var div = document.createElement('div');
+  div.setAttribute('role', 'tablist');
+  fixture.appendChild(div);
+
+  deepEqual(
+    axs.AuditRules.getRule('uncontrolledTabpanel').run({ scope: fixture }),
+    { result: axs.constants.AuditResult.NA }
+  );
+});
+
+test('Tabpanel with aria-labelledby === PASS.', function() {
+  // Setup fixture
+  var fixture = document.getElementById('qunit-fixture');
+  var tabList = document.createElement('div');
+  tabList.setAttribute('role', 'tablist');
+  fixture.appendChild(tabList);
+  var tab = document.createElement('div');
+  tab.setAttribute('role', 'tab');
+  tab.setAttribute('id', 'tabId');
+
+  tabList.appendChild(tab);
+  var tabPanel = document.createElement('div');
+  tabPanel.setAttribute('role', 'tabpanel');
+  tabPanel.setAttribute('aria-labelledby', 'tabId');
+  fixture.appendChild(tabPanel);
+
+  deepEqual(
+    axs.AuditRules.getRule('uncontrolledTabpanel').run({ scope: fixture }),
+    { elements: [], result: axs.constants.AuditResult.PASS }
+  );
+});
+
+test('Tabpanel which is controlled via aria-controls on the tab === PASS.', function() {
+  // Setup fixture
+  var fixture = document.getElementById('qunit-fixture');
+  var tabList = document.createElement('div');
+  tabList.setAttribute('role', 'tablist');
+  fixture.appendChild(tabList);
+  var tab = document.createElement('div');
+  tab.setAttribute('role', 'tab');
+  tab.setAttribute('aria-controls', 'tabpanelId');
+
+  tabList.appendChild(tab);
+  var tabPanel = document.createElement('div');
+  tabPanel.setAttribute('role', 'tabpanel');
+  tabPanel.setAttribute('id', 'tabpanelId');
+  fixture.appendChild(tabPanel);
+
```

Comment:
Should add a passing test where there is more than one tab, but only one tabpanel to ensure the false positives do not occur